### PR TITLE
EdgeSplitModifier - Support other attributes

### PR DESCRIPTION
**Pull request by Mcgode**
**Description**

`EdgeSplitModifier` now supports geometries with more attributes than just `position`.
Example has been updated to demonstrate the support of the `uv` attribute:
image

Additional attributes will also be used to split edges:
image
*Edge splits based on UV coordinates cut-off only*

This contribution is funded by [Dualbox](https://dualbox.com).

### CODE REVIEW

File: examples/js/modifiers/EdgeSplitModifier.js [hunk 1]
```diff
@@ -153,13 +152,27 @@ THREE.EdgeSplitModifier = function () {
 
 	this.modify = function ( geometry, cutOffAngle ) {
 
+		const wasNotBufferGeometry = ! geometry.isBufferGeometry;
```
Comment (mrdoob) [inline examples/js/modifiers/EdgeSplitModifier.js:155]:
```js
const wasNotBufferGeometry = geometry.isBufferGeometry === undefined;
```

File: examples/js/modifiers/EdgeSplitModifier.js [hunk 2]
```diff
@@ -153,13 +152,27 @@ THREE.EdgeSplitModifier = function () {
...
 
 		}
 
 
+		let hadNormals = false;
+		if ( geometry.attributes.normal ) {
+
+			hadNormals = true;
+
+			if ( ! wasNotBufferGeometry )
```
Comment (mrdoob) [inline examples/js/modifiers/EdgeSplitModifier.js:168]:
```js
if ( wasNotBufferGeometry === false )
```

**Comment by mrdoob** [comment]
Looking good!

Do you mind doing these two tweaks? Makes the code more robust.

**Comment by Mcgode** [comment]
Sorry about the delay. Improved the code robustness as you asked

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**